pycbc/conversions.py

```diff
@@ -427,13 +427,15 @@ def primary_xi(mass1, mass2, spin1x, spin1y, spin2x, spin2y):
     """Returns the effective precession spin argument for the larger mass.
     """
     spinx = primary_spin(mass1, mass2, spin1x, spin2x)
-    spiny = secondary_spin(mass1, mass2, spin1y, spin2y)
+    spiny = primary_spin(mass1, mass2, spin1y, spin2y)
     return chi_perp_from_spinx_spiny(spinx, spiny)
 
 def secondary_xi(mass1, mass2, spin1x, spin1y, spin2x, spin2y):
     """Returns the effective precession spin argument for the smaller mass.
     """
-    return xi2_from_mass1_mass2_spin2x_spin2y(mass1, mass2, spin2x, spin2y)
+    spinx = secondary_spin(mass1, mass2, spin1x, spin2x)
+    spiny = secondary_spin(mass1, mass2, spin1y, spin2y)
+    return xi2_from_mass1_mass2_spin2x_spin2y(mass1, mass2, spinx, spiny)
```

Comment:
>> whereas it actually depends on the x and y components of the primary's as well as the secondary component's spin

This is not true. It does only depend on the components of the secondary mass here. The top-level function is only given the primary measurements since it needs to determine which is the secondary mass.

Personally I like the explicitness of saying ``spin1x`` etc. because the function returns ``xi1`` not a generic ``xi``. The difference with functions like ``phi_from_spinx_spiny`` is those do not depend on primary/secondary sources.